tinygo: add relative and absolute --dir options to wasmtime args

ydnar opened the pull request:
This normalizes the `--dir` arguments passed to `wasmtime` for both `wasip1 `and `wasip2` when running `tinygo test` or `tinygo run`. The additional `--dir` and `PWD` arguments will now be reflected when using `-x`.

ydnar committed:
tinygo: add relative and absolute --dir options to wasmtime args

dgryski commented:
I think there is some code in the waisp1 or wasip2 branch that detects "something" (will need to double check) based on the first or last --dir option provided.  Are we breaking anything else by effectively having two names for each directory in the module path here?

ydnar commented:
> I think there is some code in the waisp1 or wasip2 branch that detects "something" (will need to double check) based on the first or last --dir option provided. Are we breaking anything else by effectively having two names for each directory in the module path here?

It worked for the tests I ran. Did you discover anything?

dgryski commented:
No, I was just wondering because I know I had issues with argument ordering.  I didn't see any failures, but I can run my larger test corpus and see what falls out.

aykevl commented:
@ydnar did you get any further with this PR?
You could try running the test locally (`make test GOTESTPKGS=. GOTESTFLAGS="-v -run=Build"`), maybe it'll give some more information as to what goes wrong?

ydnar commented:
I haven't. I'll try that!

deadprogram commented:
@ydnar looks like CI still failing here:
https://github.com/tinygo-org/tinygo/actions/runs/10874218316/job/30180223920?pr=4431#step:19:33

Possible timeout?

ydnar committed:
GNUmakefile: try 30m timeout

ydnar committed:
GNUmakefile: -timeout=1h consistent with other timeouts

ydnar committed:
GNUmakefile: go test -v

Is there a timeout or OOM issue related to the GitHub Actions runner?

ydnar committed:
GNUmakefile: go test -x

ydnar committed:
Revert "GNUmakefile: go test -x"

This reverts commit c0f6431b555ee21e1c44122106e5cfecc6e59d8e.

ydnar committed:
tinygo: only add module-relative paths to wasmtime args when running test binary

ydnar commented:
Looks like a logic error on my part. Fixed!

I wanted `-x` to accurately reflect the arguments passed to `wasmtime`, but I initially failed to account for the difference between running `go test` and `go run` in the section where I moved the `--dir` arguments to. The last commit fixes this.

deadprogram commented:
@ydnar LGTM do you want to squash these commits yourself, or would you prefer that I do it for you?

ydnar commented:
Is squash merge okay?

deadprogram commented:
> Is squash merge okay?

Done. Thank you very much!
